test/integ/test_cursor.py

```diff
@@ -1354,3 +1354,18 @@ def test_out_of_range_year(conn_cnx, result_format, cursor_type):
                     "select * from VALUES(TO_TIMESTAMP('10000-01-01 00:00:00'))"
                 )
                 cur.fetchall()
+
+
+def test_describe(conn_cnx):
+    with conn_cnx() as con:
+        with con.cursor() as cur:
+            cur.describe(
+                "select * from VALUES(1, 3.1415926, 'snow', TO_TIMESTAMP('2021-01-01 00:00:00'))"
+            )
+            assert cur.description is not None
+            column_types = [column[1] for column in cur.description]
+            assert constants.FIELD_ID_TO_NAME[column_types[0]] == "FIXED"
```

Comment:
These are the logical types. Shouldn't we return the SQL types?
https://snowflakecomputing.atlassian.net/wiki/spaces/EN/pages/3637294/Data+Type+Definitions

Reply:
I think there is no "sql types" in python connector (please correct me if I'm wrong @sfc-gh-mkeller). So I'm a little bit worried about the conversion in snowpark. But JDBC also gets the logical types first and convert them to sql types https://github.com/snowflakedb/snowflake-jdbc/blob/69f560b9370cbe5de112f259600911ccd74333bf/src/main/java/net/snowflake/client/jdbc/SnowflakeUtil.java#L129. 

So for scala we have Logical types -> sql types -> snowpark types, and for python we can simply have Logical types -> snowpark types, but it might be a little bit tricky to make the behavior consistent.

Reply:
OK, then we should have a function with the same functionality as JDBC's `extractColumnMetadata` that you pointed to. Then we'll deal the transformation in snowpark on a per-scenario basis.

Reply:
Why do we need this? Why don't we just convert Logical types to snowpark types directly in python snowpark? AFAIK, sql types are only used in ServerConnection.scala https://github.com/snowflakedb/thundersnow/search?q=java.sql.Types

Reply:
0. We can't go from FIXED to the exact type in Snowpark unless we have all the information that's available in the connector. This is evident in the scala function you shared. So the best place for this functionality is close to the source.
1. This is a publicly facing app, so the output should conform to the information (i.e. the data types) that we publish in our docs (there is no mention of FIXED [here](https://docs.snowflake.com/en/sql-reference/intro-summary-data-types.html)
2. Parity among JDBC and the python connector.

Reply:
I feel it would be a non-trivial change to python connector and we might need a further thought.